plugins/modules/proxmox_vm_info.py

```diff
@@ -153,13 +154,14 @@ def get_vms_from_cluster_resources(self):
                 msg="Failed to retrieve VMs information from cluster resources: %s" % e
             )
 
-    def get_vms_from_nodes(self, vms_unfiltered, type, vmid=None, node=None):
+    def get_vms_from_nodes(self, vms_unfiltered, type, vmid=None, name=None, node=None):
         vms = []
         for vm in vms_unfiltered:
             if (
                 type != vm["type"]
                 or (node and vm["node"] != node)
                 or (vmid and int(vm["vmid"]) != vmid)
+                or (name and vm["name"] != name)
```

Comment:
Can `name` be an empty string? I.e. should this line rather be `or (name is not None and ...)`?